app/src/main/java/com/example/polina/socialnetwork/ProfileActivity.java

```diff
@@ -54,38 +56,39 @@
     public ListView postList;
 
 
-    String profileURL;
-    String connection_faild;
-    SharedPreferences sharedPreferences;
-    SharedPreferences sharedPreferencesUserId;
-    public static final String PROFILE_PREFERENCES = "profile info";
-    public static final String USER_ID_PREFERENCES = "User ID";
-    public static final String USER_ID = "_id";
-    private static final String POSTS_JSON = "posts";
-    ImageLoader mImageLoader;
-
-    String iduser;
-
-
+    private String profileURL;
+    private String connectionFaild;
+    private SharedPreferences sharedPreferences;
+    private SharedPreferences sharedPreferencesUserId;
+    private ImageLoader mImageLoader;
+    private String idUser;
+    private String idPost;
     private PostAdapter adapter;
 
+    private static final int NORMAL_LIST_SIZE = 9;
+    private int totalPost = 10;
+    private ArrayList<Post> postsToLoad;
+    private int count = 0;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
+        postsToLoad = new ArrayList<>();
+        ImageLoader.ImageCache imageCache = new BitmapLruCache();
+        mImageLoader = new ImageLoader(Volley.newRequestQueue(this), imageCache);
+        connectionFaild = getResources().getString(R.string.connection_faild);
     }
 
     @Override
     protected void onResume() {
         super.onResume();
-        connection_faild = getResources().getString(R.string.connection_faild);
-        sharedPreferences = getSharedPreferences(PROFILE_PREFERENCES, MODE_PRIVATE);
-        sharedPreferencesUserId = getSharedPreferences(USER_ID_PREFERENCES, MODE_PRIVATE);
-        iduser = sharedPreferencesUserId.getString(USER_ID, "");
-        ImageLoader.ImageCache imageCache = new BitmapLruCache();
-        mImageLoader =  new ImageLoader(Volley.newRequestQueue(this), imageCache);
+        sharedPreferences = getSharedPreferences(Utils.PROFILE_PREFERENCES, MODE_PRIVATE);
+        sharedPreferencesUserId = getSharedPreferences(Utils.USER_ID_PREFERENCES, MODE_PRIVATE);
+        idUser = sharedPreferencesUserId.getString(Utils.USER_ID, "");
         adapter = new PostAdapter(this, new ArrayList<Post>(), mImageLoader);
-        if (sharedPreferences.contains(ServerAPI.NAME)) {
+        postList.setAdapter(adapter);
```

Comment:
i don't see any reason to do postList.setAdapter, setOnScrollListener each time on onResume